Remove the global repo flag

probablycorey opened the pull request:
The `--repo` flag is helpful for some commands but not others. This removes it as a global command and adds a helper function to add it to commands that need it. Here is the list.


## 🟢 Commands with the `--repo` flag
```
issue create
issue view
issue close
issue reopen
issue status
issue list

pr diff
pr review
pr status
pr close
pr reopen
pr merge
pr ready
pr list
pr view
```

## 🔴 Commands without the `--repo` flag
```
pr create
pr checkout

repo clone
repo create
repo fork
repo view

alias
api
completion
config
gist
```

Closes https://github.com/cli/cli/issues/599

probablycorey committed:
Remove global repo flag

probablycorey committed:
Add repo flag to issue commands

probablycorey committed:
Add repoflag to pr commands

probablycorey committed:
Correct package name

probablycorey commented:
I thought I could remove the repo flag from `pr checkout` but there is a test that relies on it. But I couldn't think of an example of why you would want to use `pr checkout --repo OWNER/NAME` vs `pr checkout OWNER:NAME`

probablycorey committed:
Ignore repo flag errors in determineBaseRepo

mislav reviewed (changes requested):
1. Would we extend the `--repo` flag to `pr create` and `pr checkout` as well? Specifically for `pr create`, we've been telling people that they can choose the base repository for the PR by using `--repo` https://github.com/cli/cli/issues/350#issuecomment-605988260. If we remove that now, it might break functionality that people rely on.

2. If all `issue` and `pr` commands support `--repo`, then it might be simpler to just add the flag to `issueCmd.PersistentFlags()` and `issueCmd.PersistentFlags()`, and all nested commands will inherit it by nature of "persistent flags" (just like they used to inherit it from RootCmd).

mislav committed:
Merge remote-tracking branch 'origin' into remove-global-repo-flag-for-real-this-time

mislav committed:
Declare `-R, --repo` flag on all `issue` and `pr` subcommands

mislav reviewed (approved):
Pushed some simplifications

mislav committed:
:nail_polish: be clearer about the value passed to ResolveRemotesToRepos

`repo` will always be blank here, so replace the argument with a blank
literal instead.
